pkg/common/array_arg_test.go

```diff
@@ -0,0 +1,225 @@
+package common
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+)
+
+func TestExpandArrayParameters(t *testing.T) {
+
+	setupTestCommands := func() (*cobra.Command, *cobra.Command) {
+		// Clear existing array params
+		arrayParamsInCommands = map[*cobra.Command][]string{}
+
+		rootCmd := &cobra.Command{Use: "root"}
+		subCmd := &cobra.Command{Use: "subcmd"}
+		rootCmd.AddCommand(subCmd)
+
+		recordArrayParamForCommand(subCmd, "--array-param")
+		recordArrayParamForCommand(subCmd, "-a")
+		recordArrayParamForCommand(subCmd, "--multi")
+
+		return rootCmd, subCmd
+	}
+
+	t.Run("should handle command without array parameters", func(t *testing.T) {
+		g := NewWithT(t)
+
+		argv := []string{"root", "command", "--param", "value"}
+		result := ExpandArrayParameters(argv)
+
+		expected := []string{"root", "command", "--param", "value"}
+		g.Expect(result).To(Equal(expected))
+	})
+
+	t.Run("should handle non-array parameters normally", func(t *testing.T) {
+		g := NewWithT(t)
+
+		setupTestCommands()
+
+		argv := []string{"root", "subcmd", "--normal-param", "value", "--other-flag"}
+		result := ExpandArrayParameters(argv)
+
+		expected := []string{"root", "subcmd", "--normal-param", "value", "--other-flag"}
+		g.Expect(result).To(Equal(expected))
+	})
+
+	t.Run("should expand array parameters with multiple values", func(t *testing.T) {
+		g := NewWithT(t)
+
+		setupTestCommands()
+
+		argv := []string{"subcmd", "--array-param", "val1", "val2", "val3", "--other-flag"}
+		result := ExpandArrayParameters(argv)
+
+		expected := []string{"subcmd", "--array-param", "val1", "--array-param", "val2", "--array-param", "val3", "--other-flag"}
+		g.Expect(result).To(Equal(expected))
+	})
+
+	t.Run("should expand short form array parameters", func(t *testing.T) {
+		g := NewWithT(t)
+
+		setupTestCommands()
+
+		argv := []string{"subcmd", "-a", "val1", "val2", "--other-flag"}
+		result := ExpandArrayParameters(argv)
+
+		expected := []string{"subcmd", "-a", "val1", "-a", "val2", "--other-flag"}
+		g.Expect(result).To(Equal(expected))
+	})
+
+	t.Run("should handle array parameters with equals syntax", func(t *testing.T) {
+		g := NewWithT(t)
+
+		setupTestCommands()
+
+		argv := []string{"subcmd", "--flag", "--array-param=val1", "val2", "val3", "--other-flag"}
+		result := ExpandArrayParameters(argv)
+
+		expected := []string{"subcmd", "--flag", "--array-param", "val1", "--array-param", "val2", "--array-param", "val3", "--other-flag"}
```

Comment:
For the `--array=value` case, I would expect the array to only have the one value. It looks like a clear attempt to separate the remaining values from the array

```suggestion
		argv := []string{"subcmd", "--flag", "--array-param=val1", "val2", "val3", "--other-flag"}
		result := ExpandArrayParameters(argv)

		expected := []string{"subcmd", "--flag", "--array-param", "val1", "val2", "val3", "--other-flag"}
```